src/runtime_src/hip/core/module.cpp

```diff
@@ -8,50 +8,37 @@
 
 namespace xrt::core::hip {
 void
-module::
+module_xclbin::
 create_hw_context()
 {
-  auto xrt_dev = m_ctx->get_xrt_device();
+  auto xrt_dev = get_context()->get_xrt_device();
   auto uuid = xrt_dev.register_xclbin(m_xclbin);
   m_hw_ctx = xrt::hw_context{xrt_dev, uuid};
 }
 
-module::
-module(std::shared_ptr<context> ctx, const std::string& file_name)
-  : m_ctx{std::move(ctx)}
+module_xclbin::
+module_xclbin(std::shared_ptr<context> ctx, const std::string& file_name)
```

Comment:
```suggestion
module_xclbin(const std::shared_ptr<context>& ctx, const std::string& file_name)
```
Either move into module ctor or receive by const ref and let module ctor move as it already does.